model/waterfall.go

```diff
@@ -224,3 +228,38 @@ func GetWaterfallBuildVariants(ctx context.Context, versionIds []string) ([]Wate
 
 	return res, nil
 }
+
+// GetNextRecentActiveWaterfallVersion returns the next recent active version on the waterfall, i.e., a newer
+// version that is activated.
+func GetNextRecentActiveWaterfallVersion(ctx context.Context, projectId string, maxOrder int) (*Version, error) {
```

Comment:
If we're looking for a version with an order > `maxOrder`, I'd suggest the parameter be renamed to `minOrder`